### PR TITLE
Postgre sql server BRS

### PR DESCRIPTION
**Pull request by pranchalsomani**

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 [hunk 1]
```diff
@@ -0,0 +1,903 @@
...
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+  
+    Write-Host "Found [$($totalDBForPostgreSQLFS)] Azure Database for PostgreSQL flexible server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list for storing Azure Database for PostgreSQL flexible server(s) for which server parameter require_secure_transport in not ON
+    $DBForPostgreSQLFSWithNonSecureTLSVersionEnabled = @()
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:369]: This is holding resources non-compliant for both require_secure_transport and ssl_min_protocol_version right? Comment and variable are not inline to this

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 [hunk 2]
```diff
@@ -0,0 +1,903 @@
...
+            $logResource.Add("ResourceName", ($_.ResourceName))	
+            $logResource.Add("Reason", "TLS version(s) and secure transport configured on this Resource.")    	
+            $logSkippedResources += $logResource	
+        }
+    }
+   
+    $totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled = ($DBForPostgreSQLFSWithNonSecureTLSVersionEnabled  | Measure-Object).Count
+
+    if ($totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled -eq 0) {
+        Write-Host "No Azure Database for PostgreSQL flexible server(s) found with non-secure TLS version enabled and secure transport communication is disabled.. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:389]: NIT: Message is not intuitive. **and** to be replaced by **or**.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 [hunk 3]
```diff
@@ -0,0 +1,903 @@
...
+                    $logControl.RemediatedResources = $logRemediatedResources
+                    $logControl.SkippedResources = $logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+        return
+    }
+
+    Write-Host "Found [$($totalDBForPostgreSQLFSWithNonSecureTLSVersionEnabled )] Azure Database for PostgreSQL flexible server(s) for which non secure TLS version is enabled ." -ForegroundColor $([Constants]::MessageType.Update)
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:406]: Is this just for non secure TLS version is enabled and not the other parameter?

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 [hunk 4]
```diff
@@ -0,0 +1,903 @@
...
+        Write-Host "Connecting to Azure account..."
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else {
+        # Setting up context for the current Subscription.
+        $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    }
+
+
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:693]: NIT: Too may new lines

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 [hunk 5]
```diff
@@ -0,0 +1,903 @@
...
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+  
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 3] Set TLS version and secure transport communication to previous value on all Azure Database for PostgreSQL flexible server(s) in the Subscription"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ( -not $Force) {
+
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:756]: NIT: Indent the file and extra spaces

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 [hunk 6]
```diff
@@ -0,0 +1,903 @@
...
+    $nonCompliantTLSVersions = @();
+    try {
+        $supportedTLSVersions = $TLSVersion.ToLower().Replace([Constants]::TLSversionPattern, "").Split(',');
+        $supportedTLSVersions | ForEach-Object {
+            if ([double] $_ -lt [Constants]::MinRequiredTLSVersion) {
+                $nonCompliantTLSVersions += $_
+            }
+        }
+    }
+    Catch {
```
Comment (vaishnavipulluri) [inline Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:859]: NIT: Catch with uppercase C

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 (hunk 2 above)
Comment (pranchalsomani) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:389]: Fixed

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 (hunk 4 above)
Comment (pranchalsomani) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:693]: Fixed

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 (hunk 3 above)
Comment (pranchalsomani) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:406]: Fixed

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 (hunk 1 above)
Comment (pranchalsomani) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:369]: Fixed.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 (hunk 5 above)
Comment (pranchalsomani) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:756]: Fixed.

File: Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1 (hunk 6 above)
Comment (pranchalsomani) [reply Scripts/RemediationScripts/Remediate-SetDBForPostgreSQLFlexibleServerMinReqTLSVersion.ps1:859]: Fixed.

**Review by vaishnavipulluri — APPROVED** [review]

**Review by Aboli-msft — APPROVED** [review]

**Pull request merged**